plugins/chat/assets/javascripts/discourse/components/modal/chat/archive-channel.js

```diff
@@ -95,14 +66,44 @@ export default Component.extend(ModalFunctionality, {
       "chat.selection.existing_topic.instructions_channel_archive"
     );
     return labels;
-  },
+  }
 
-  @discourseComputed()
-  instructionsText() {
+  get instructionsText() {
     return htmlSafe(
       I18n.t("chat.channel_archive.instructions", {
-        channelTitle: this.chatChannel.escapedTitle,
+        channelTitle: this.channel.escapedTitle,
       })
     );
-  },
-});
+  }
+
+  @action
+  archiveChannel() {
+    this.saving = true;
+
+    return this.chatApi
+      .createChannelArchive(this.channel.id, this.#data())
+      .then(() => {
+        this.flash(I18n.t("chat.channel_archive.process_started"), "success");
+        this.channel.status = CHANNEL_STATUSES.archived;
+
+        discourseLater(() => {
+          this.closeModal();
```

Comment:
```suggestion
          this.args.closeModal();
```

(`this.closeModal` was previously provided by the `ModalFunctionality` mixin)